src/ethereum/frontier/vm/interpreter.py

```diff
@@ -93,7 +102,9 @@ def process_call(
             raise InvalidOpcode(evm.code[evm.pc])
 
         op_implementation[op](evm)
-        evm.pc += 1
+
+        if op not in PC_CHANGING_OPS:
+            evm.pc += 1
```

Comment:
I don't think this kind of "do something _except_ in certain cases" logic is the best for the specification.

I know it would be more tedious, but what would you think of lowering the `evm.pc += 1` into each opcode's implementation function? I think it would make it immediately obvious how each opcode behaves, at the risk of repeating ourselves.

Reply:
While development, I felt that an argument can come in support either ways and hence went ahead with this as it seemed easy to revert.

Moving evm.pc increment to each opcodes kind of makes the workflow a bit obscure and not so straight forward. But moving it actually makes the design more streamlined. Let me know if the obscurity is worth it. Thanks.

Reply:
In the interest of avoiding another round trip, I'll merge as is, but this'll be a candidate for changing later on.

Reply:
Logged #278 for this.